editor/views/editoritem.py

```diff
@@ -511,10 +511,51 @@ def get(self, *args, **kwargs):
 
     def post(self, request, *args, **kwargs):
         ei = self.get_object()
+        error_message = ''
+        count = 1
+        if (ei.item_type == 'exam'):
+            exam_questions = ei.exam.newexamquestion_set.all().order_by('qn_order').distinct()
+            for eq in exam_questions:
+                question = eq.question
+                if question.editoritem.published == True:
+                    continue
+                current_error_message = ''
+                if not question.editoritem.can_be_edited_by(self.request.user):
+                    current_error_message = 'User cannot edit this question. '
+                edit_content = question.editoritem.edit_dict()['JSONContent']
+                if edit_content['statement'] == '':
```

Comment:
I'm not very happy with these things being checked here - checking if something's ready to be published is something you might want to do in other circumstances, so we should try to avoid repeating code.
I think NewQuestion should have a method like `ready_to_publish`, which either returns `True` or throws an error with a message describing the reason.